content/docs/concepts/low-power.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "Low Power"
+linkTitle: "Low Power"
+type: "docs"
+weight: 35
+description: >
+  How to consume less power on microcontrollers
+---
+
+Microcontrollers are usually rather non-demanding when it comes to power consumption.
+Couple of watts is the most you may expect a hobby project would use.
+
+Sometimes, even that is not low enough.
+Power your project from a battery, and it'd benefit greatly from consuming less.
+
+Common scenario for such projects is sleeping most of the time, and only rarely briefly wake up to query sensors and execute actions.
+The main caveat here is if not configured properly, microcontroller and peripherals continue to consume power as usual while your program "sleeps".
+
+The standard way to pause execution in Go is to use `time.Sleep()` function that is available in TinyGo too. Depending on the microcontroller architecture and effort put into it, the efficiency of this function can vary. The only thing certain: this is better than busy looping until a certain time has passed.  
+Still, with such high-level function as `time.Sleep()` it is not always possible to disable everything that consumes power as the decisions may depend on the application or the board used.
+
+Most microcontrollers have one or more "low power" modes, also known as "deep sleep" and, sometimes, "dormant".  
+In such modes microcontroller may consume as low as milli- or even micro-Watts.  
+TinyGo gives you low level access to hardware to try and make most of it.
+
+## Implementations
+
+Manufacturers implement different modes, depending of capabilites of their products and name these modes differently too.
+Hence no naming consistency across microcontrollers, unfortunately.
+
+Always consult "Power consumption" part of a respective datasheet to understand capabilities of your hardware.  
+When reading datasheets, keep in mind peripherals that draw power too. These include \[power\] LEDs, various sensors or co-processors usually present on development boards.
+
+Capabilities of different chips:
+- Some very good chips (like the nRF series) can go to a very low power state (just a few µA) with RAM retention and waking up from a low-power RTC.
+- Some limited chips (like the esp32) can't really "sleep" with low power -- they can disable the core and memory and reset after a predefined time losing state stored in memory.
+- Other chips need to use a special feature to go into deep sleep mode.
+
+Especially the distinction between waking up with or without RAM is important, because it greatly affects how you write software.
+
+The common way to wake up from deep sleep is to use [interrupts](compiler-internals/interrupts), such as timer, real-time-clock or an interrupt on a pin change.
+
+### RP2040
+
+Minimal achievable power consumption according to [RP2040 datasheet](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf): 6mW to 12mW (0.19mA at 3.3V)
```

Comment:
It would be nice to link to a specific page in the datasheet (add `#page=xxx` to the end of the URL) and/or also specify the section where this is specified.

Reply:
Good, will do.